### DISCUSSION TITLE
목데이터를 제공하는 api를 먼저 구현해야할까?

### DISCUSSION THREAD

**Discussion by 5witchkr** [ProjectCoco/wayfarer-backend#3] [Brainstorming]
백엔드의 개발은 TDD를 적용하기로 했습니다.

하지만 프론트엔드팀의 개발일정을 맞추려면 목데이터를 반환해줄 필요성을 느끼는데

서비스인터페이스를 정의하고 목데이터를 반환해주는 구현체를 하나 만들어서 목데이터를 제공하고자하는데

어떻게 생각하시나요?

**Comment by Minoolian** [comment]
서비스 레이어를 추상화하여 테스트 기간에는 Mock 데이터를 반환하는 구현클래스, 서비스 기간에는 실제 데이터를 반환하는 구현클래스를 주입하여 사용하면 기존의 Controller 코드 변경없이 기능을 변경할 수 있을 것 같습니다. (OCP)

**Reply by 5witchkr** [reply]
그럼 목데이터를 반환해주는 api를 먼저 만들어보는 방향으로 가시죠 !!
프론트엔드분들도 실제 api에 바로 연결해보면 되니까
목서비스워커를 사용하지 않아도 돼서  편할것같다는 긍정적인 의견이 있었어요.
